Formula/rust.rb

```diff
@@ -79,10 +86,30 @@ def install
       system "make", "install"
     end
 
+    if build.with? "racer"
+      resource("racer").stage do
+        (buildpath/"cargo_home").mkdir
+        ENV["PATH"] = ENV["PATH"]+":"+bin
+        ENV["CARGO_HOME"] = buildpath/"cargo_home"
+        # interactive_shell
+        system (bin/"cargo"), "build", "--release", "--verbose"
+        (libexec/"bin").install "target/release/racer"
+        (bin/"racer").write_env_script(libexec/"bin/racer", :RUST_SRC_PATH => pkgshare/"rust_src/current")
```

Comment:
actually, even opt_pkgshare shouldn't be needed since the sources are living together with rust here. seems like you can use the actual path, no?